algorithms/closest_pair.py

```diff
@@ -84,7 +84,12 @@ def _distance_between(point_1, point_2):
 
```

Comment:
I'm not sure I understand this bit https://github.com/Wynndow/algorithms/pull/2/files#diff-01016d6362c2e04e50a730472494b7c2R46

```python
def _calculate_closest_split_pair(x_sorted, y_sorted, delta):
     mid_x = x_sorted[len(x_sorted)//2][0]		     mid_x = x_sorted[len(x_sorted)//2][0]
     potential_closest_split_pairs = [		     potential_closest_split_pairs = [
         point for point in y_sorted		         point for point in y_sorted
         if point[0] >= mid_x - delta		         if point[0] >= mid_x - delta
         and point[0] <= mid_x + delta		         and point[0] <= mid_x + delta
     ]		     ]
```

What's `y_sorted for`? I may be wrong (I just scanned the algorithm on [wikipedia](https://en.wikipedia.org/wiki/Closest_pair_of_points_problem)), but I think it should be `x_sorted`, and you don't need `y_sorted` anywhere.

The first time you call `recurse_to_answer`, `x_sorted` and `y_sorted` contain the same points. Then you split it into two sets, but the two lists are divided differently. The first time you recurse, `recurse_to_answer` will get called with the `x_sorted` as the left half, and `y_sorted` as the bottom half. So `potential_closest_split_pairs` is filtering one list based on a completely different one.

With `x_sorted` I think this works as you expect and you'd get all the points that are within `delta` of the midpoint.